pkg/appliance/backup/backup.go

```diff
@@ -86,6 +90,73 @@ func (b *Backup) Download(ctx context.Context, applianceID, backupID, destinatio
 	return out, nil
 }
 
+// Download a completed appliance backup with the given ID of an Appliance
+func (b *Backup) Download(ctx context.Context, applianceID, backupID, destination string) (*os.File, error) {
+	out, err := os.Create(destination)
+	if err != nil {
+		return nil, err
+	}
+	defer out.Close()
+
+	client := b.HTTPClient
+	cfg := b.APIClient.GetConfig()
+	url, err := cfg.ServerURLWithContext(ctx, "ApplianceBackupApiService.AppliancesIdBackupBackupIdGet")
+	if err != nil {
+		return nil, err
+	}
+	url = fmt.Sprintf("%s/appliances/%s/backup/%s", url, applianceID, backupID)
+	ctx = context.WithValue(ctx, api.ContextAcceptValue, fmt.Sprintf("application/vnd.appgate.peer-v%d+gpg", b.Version))
+
+	headReq, err := http.NewRequestWithContext(ctx, http.MethodHead, url, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+	head, err := client.Do(headReq)
+	if err != nil {
+		return nil, err
+	}
+
+	size := head.ContentLength
+	if err := retryDownload(ctx, client, out, url, size); err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+func retryDownload(ctx context.Context, client *http.Client, f *os.File, url string, size int64) error {
+	start := int64(0)
+	return backoff.Retry(func() error {
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+		if err != nil {
+			return backoff.Permanent(err)
+		}
+		req.Header.Set("Range", fmt.Sprintf("bytes=%d-", start))
+
+		log := logrus.WithField("request", req)
+		res, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		log.WithField("response", res).Debug("download request and response")
```

Comment:
Will that log the range header?

Reply:
Verified log.
Im being really picky, but the colors don't work on DEBUG log for some reason. 

<img width="1791" alt="Screenshot 2024-01-25 at 8 45 21 AM" src="https://github.com/appgate/sdpctl/assets/7606627/bf2d09bb-03ef-46ed-a199-79567b8eca90">

Reply:
yep

Reply:
> Verified log. Im being really picky, but the colors don't work on DEBUG log for some reason.

It shows correct for me though. Maybe your shell is just rendering `DEBUG` in the same color as the default?